v2/internal/genericarmclient/generic_client.go

```diff
@@ -84,6 +84,16 @@ func NewGenericClientFromHTTPClient(cloudCfg cloud.Configuration, creds azcore.T
 	// the value IN the interface IS nil).
 	if httpClient != nil {
 		opts.Transport = httpClient
+	} else {
+		// If httpClient is not provided, we use a HTTPClient with default Transport + settings
+		// to establish multiple TCP ARMClient connections to avoid throttling.
+		// TODO: Use https://github.com/Azure/go-armbalancer here once its prod ready.
```

Comment:
IMO we should construct a default HTTPClient of our own similar to this: https://github.com/kubernetes-sigs/cloud-provider-azure/blob/c33f413c0ef58a7c7af4bdaed87fd1a6e4714f99/pkg/azureclients/armclient/azure_armclient.go#L70

Otherwise we're:
1. Modifying the default HTTPClient, which seems bad?
2. We aren't' actually using the default settings that the SDK uses, we should use basically their default settings but with some modifications. Possibly we could clone their transport (if it's accessible, not sure if it is).